src/variety/cbblock.js

```diff
@@ -51,6 +98,20 @@ Board:{
 	}
 },
 
+"Cell@dbchoco": {
+	maxnum: function() {
+		var bd=this.board;
+		return (bd.cols*bd.rows)>>1;
+	},
+
+	posthook: {
+		ques: function(num) {
+			 this.board.tilegraph.rebuild();
+			 this.board.blockgraph.rebuild();
+		}
+	}
+},
+
 "AreaTileGraph:AreaGraphBase":{
 	enabled : true,
 	relation : {'border.ques':'separator'},
```

Comment:
(let's see if commenting on an unchanged line works)

It would appear more correct to have `relation : {'border.qans':'separator'}` here for `dbchoco`. In principle, this should cause graph rebuilds when `border.qans` changes (though it's a bit unclear that can really be made to work correctly). Worth a try though.